packages/dialog/src/index.js

```diff
@@ -179,3 +201,27 @@ function useForkedRef(...refs) {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, refs);
 }
+
+function validateAriaLabel(props, name, compName, ...rest) {
+  const details =
+    "\nSee https://www.w3.org/TR/wai-aria/#aria-label for details.";
+  if (!props["aria-label"] && !props["aria-labelledby"]) {
+    return new Error(
+      "You must provide either an `aria-label` prop or an `aria-labelledby` prop to the `" +
+        compName +
+        "` component." +
+        details
+    );
+  }
+  if (props["aria-label"] && props["aria-labelledby"]) {
+    return new Error(
+      "You provided both `aria-label` and `aria-labelledby` props to the `" +
+        compName +
+        "` component. If the a label for this component is visible on the screen, that label's component should be given a unique ID prop, and that ID should be passed as the `aria-labelledby` prop into `" +
```

Comment:
Same here. Let's say "You provided both \`aria-label\` and \`aria-labelledby\` props to a \<Dialog>"